### PR TITLE
[Core][MPI] ParallelFillComm - adding checks that make debugging easier

### PR DESCRIPTION
**Pull request by philbucher**
I tried creating a distributed modelpart in python by hand (without Metis) and ofc I forgot assigning `PARTITION_INDEX` (:man_facepalming:), which leads to amazing errors like `Calling the default constructor for the node ... illegal operation!!`. What a joy to debug...

Long story short, this should hopefully prevent this in the future ...

### CODE REVIEW

**Review by roigcarlo — DISMISSED** [review]
To me looks ok

File: kratos/mpi/utilities/parallel_fill_communicator.cpp [hunk 1]
```diff
@@ -194,19 +194,36 @@ void ParallelFillCommunicator::PrintData(std::ostream& rOStream) const
...
+    int non_zero_partition_index_found = 0;
+    for (const auto& r_node : rModelPart.Nodes()) {
+        const int node_partition_index = r_node.FastGetSolutionStepValue(PARTITION_INDEX);
+        if (node_partition_index != 0) {
+            non_zero_partition_index_found = 1;
+            break;
+        }
+    }
+
+    pnew_comm->SumAll(non_zero_partition_index_found);
```
Comment (jcotela) [inline kratos/mpi/utilities/parallel_fill_communicator.cpp:220]: Note that DataCommunicator should have an OrAll option for bool (and, if it is not there yet, I probably should add it)

**Review by jcotela — DISMISSED** [review]

**Comment by rubenzorrilla** [comment]
Fine for me as well.

File: kratos/mpi/utilities/parallel_fill_communicator.cpp (hunk 1 above)
Comment (philbucher) [reply kratos/mpi/utilities/parallel_fill_communicator.cpp:220]: I suspected that there would be a better solution, but I didn't find it, could you point to what you would do?

Comment (jcotela) [reply kratos/mpi/utilities/parallel_fill_communicator.cpp:220]: The `DataCommunicator::OrReduceAll` method would fit nicely here, unfortunately I just realized that it is part of #5929. I can try separating it into a smaller PR, but I would not block this PR for that.

**Review by jcotela — DISMISSED** [review]
Re-approving

**Comment by philbucher** [comment]
can someone reapprove pls?

**Review by roigcarlo — APPROVED** [review]

**Comment by rubenzorrilla** [comment]
@philbucher I think you forgot to merge this one (too many PRs :laughing: ).

**Comment by philbucher** [comment]
Thx
Too much stuffs lately

**Pull request merged**